# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmMemory.h
# Commit: 75d829afb4b0
# Full Hash: 75d829afb4b06a3a8b7e91ea15c8a0cf34665fb7
# Author: Cosmin Sabou <csabou@mozilla.com>
# Date: 2025-09-16 12:02 +0300
# Regressor Bug: 1969509
# File Overlap Count: 7
# Overlapping Files: js/src/vm/ArrayBufferObject.cpp, js/src/wasm/WasmMemory.h, js/src/jit-test/tests/wasm/wasm-resizablearraybuffer-shared.js, js/src/jit-test/tests/wasm/memory64/basic.js, js/src/jit-test/tests/wasm/wasm-resizablearraybuffer.js
# Description:
#   Revert "Bug 1969509 - wasm: Update .maxByteLength to report the source wasm max pages in bytes. r=rhunt" for causing javascript engine crashes (bug 1988694)
# ==============================================================================

diff -r 896c8ee9171d -r 75d829afb4b0 js/src/wasm/WasmMemory.h
--- a/js/src/wasm/WasmMemory.h	Tue Sep 16 04:43:56 2025 +0300
+++ b/js/src/wasm/WasmMemory.h	Tue Sep 16 12:02:13 2025 +0300
@@ -102,13 +102,6 @@
     return length.value();
   }
 
-  // Return the byteLength for a 64-bits memory.
-  uint64_t byteLength64() const {
-    mozilla::CheckedInt<uint64_t> length(value_);
-    length *= PageSize;
-    return length.value();
-  }
-
   // Increment this pages by delta and return whether the resulting value
   // did not overflow. If there is no overflow, then this is set to the
   // resulting value.
